verilog/dv/cocotb/run.py

```diff
@@ -0,0 +1,303 @@
+                    self.passed_tests +=1
+                elif test_run.passed == "failed":
+                    self.failed_tests +=1
+                self.unknown_tests -=1
+                self.update_reg_log()
+        #TODO: add send mail here
+    
+
+    def update_reg_log(self):
+        file_name=f"sim/{os.getenv('RUNTAG')}/runs.log"
+        f = open(file_name, "w")
+        f.write(f"{'Test':<25} {'status':<10} {'start':<15} {'end':<15} {'duration':<13} {'p/f':<5}\n")
+        for test,sim_types in self.tests.items():
+            for sim_type,status in sim_types.items():
+                new_test_name= f"{sim_type}-{test}"
+                f.write(f"{new_test_name:<25} {status['status']:<10} {status['starttime']:<15} {status['endtime']:<15} {status['duration']:<13} {status['pass']:<5}\n")
+        f.write(f"\n\nTotal: ({self.passed_tests})passed ({self.failed_tests})failed ({self.unknown_tests})unknown ")
+        f.close()
+    
+    def write_command_log(self):
+        file_name=f"sim/{os.getenv('RUNTAG')}/command.log"
+        f = open(file_name, "w")
+        f.write(f"{' '.join(sys.argv)}")
+        f.close()
+
+class main():
+    def __init__(self,args) -> None:
+        self.regression = args.regression
+        self.test       = args.test
+        self.testlist   = args.testlist
+        self.type       = args.sim
+        self.tag        = args.tag
+        self.maxerr        = args.maxerr
+        self.check_valid_args()
+        self.set_tag()
+        self.def_env_vars()
+        RunRegression(self.regression,self.test,self.type,self.testlist)
+
+    def check_valid_args(self):
+        if all(v is  None for v in [self.regression, self.test, self.testlist]):
+            print ("Fatal: Should provide at least one of the following options regression, test or testlist for more info use --help")
+            sys.exit()
+        if not set(self.type).issubset(["RTL","GL","GL_SDF"]):
+            print (f"Fatal: {self.type} isnt a correct type for -sim it should be one or combination of the following RTL, GL or GL_SDF")
+            sys.exit()
+    def set_tag(self):
+        self.TAG = None # tag will be set in the main phase and other functions will use it
+        if self.tag is not None:
+            self.TAG = self.tag
+        elif self.regression is not None: 
+            self.TAG = f'{self.regression}_{datetime.now().strftime("%H_%M_%S_%d_%m")}'
+        else: 
+            self.TAG = f'run{random.randint(0,1000)}_{datetime.now().strftime("%H_%M_%S_%d_%m")}'
+        Path(f"sim/{self.TAG}").mkdir(parents=True, exist_ok=True)
+        print(f"Run tag: {self.TAG}")
+
+    def def_env_vars(self):
+        cocotb_path = os.getcwd()
+        repo_path = go_up(cocotb_path,4)
+        os.environ["CARAVEL_ROOT"] = f"{repo_path}/caravel"
+        os.environ["CARAVEL_VERILOG_PATH"] = f"{repo_path}/caravel/verilog"
+        os.environ["MCW_ROOT"] = f"{repo_path}/caravel_mgmt_soc_litex/"
```

Comment:
Here you are assuming a specific directory structure, so either you write down what that structure is, and state that it's a dependency, or you do something to make sure that these files exist, or do something to start the setup (something like `make setup` in caravel_user_project)

Reply:
usually `caravel_mgmt_soc_litex` is renamed to `mgmt_core_wrapper`

Reply:
There also should be an option to change the `MCW_ROOT` , so that if I type in the cmd `export MCW_ROOT=<dir path>` I can run it from wherever I want